src/lib/viewers/archive/ArchiveExplorer.js

```diff
@@ -0,0 +1,161 @@
+        const folders = [];
+        const temp = {};
+        const cache = {};
+        let root;
+
+        itemList.forEach(info => {
+            if (!info.parent) {
+                root = info.absolute_path;
+            }
+            if (info.type === 'folder') {
+                folders.push(info);
+            }
+            temp[info.absolute_path] = info;
+        });
+
+        folders.forEach(folderInfo => {
+            cache[folderInfo.absolute_path] = folderInfo.item_collection.entries.map(item => temp[item.absolute_path]);
+        });
+
+        return { cache, root };
+    });
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            fullPath: undefined,
+        };
+    }
+
+    getRowData = itemList => ({ index }) => {
+        const { name, type, modified_at: modifiedAt, ...rest } = itemList[index];
+
+        const rowData = {
+            [KEY_NAME]: {
+                isExternal: false,
+                name,
+                type,
+            },
+            [KEY_MODIFIED_AT]: `20${modifiedAt}`,
+            ...rest,
+        };
+        return rowData;
+    };
+
+    handleClick = cellValue => {
+        const { name } = cellValue;
+        const { fullPath } = this.state;
+        const nextFullPath = `${fullPath}${name}/`;
+        this.setState({
+            fullPath: nextFullPath,
+        });
+    };
+
+    render() {
+        const { itemCollection } = this.props;
+        const { fullPath } = this.state;
+        const { cache, root } = this.buildCache(itemCollection);
+
+        if (!fullPath) {
+            this.setState({
+                fullPath: root,
+            });
+            return null;
+        }
+
+        const itemList = cache[fullPath];
+
+        return (
+            <Internationalize language="en-us" messages={{}}>
+                <VirtualizedTable
+                    className="ArchiveFilesTable"
+                    rowData={itemList}
+                    rowGetter={this.getRowData(itemList)}
+                >
+                    {intl => [
+                        <Column
+                            key="Filename"
+                            cellRenderer={itemNameCellRenderer(intl, this.handleClick)}
+                            dataKey={KEY_NAME}
+                            disableSort
+                            flexGrow={3}
+                            label={__('filename')}
+                            width={1}
+                        />,
+                        <Column
+                            key="Last modified date"
+                            cellRenderer={readableTimeCellRenderer}
+                            dataKey={KEY_MODIFIED_AT}
+                            disableSort
+                            flexGrow={2}
+                            label={__('last_modified_date')}
+                            width={1}
+                        />,
+                        <Column
+                            key="Size"
```

Comment:
Should these keys be lowercase/hyphenated? Should they use the column key strings to avoid duplication?